src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Remote/RemoteServiceInvoker.cs

```diff
@@ -79,8 +79,13 @@ internal sealed class RemoteServiceInvoker(
         }
     }
 
-    private async Task<RazorRemoteHostClient?> TryGetClientAsync(CancellationToken cancellationToken)
+    private async Task<RazorRemoteHostClient?> TryGetClientAsync(bool doInit, CancellationToken cancellationToken)
     {
+        if (!_fullyInitialized && doInit)
```

Comment:
The comment below about initialising should probably be moved here. Comments are good.

Reply:
Moved and modified to make sense for the new client roles